agent/agent.go

```diff
@@ -684,6 +692,26 @@ func (a *Agent) Start(ctx context.Context) error {
 		{Name: "pre_release", Value: a.config.VersionPrerelease},
 	})
 
+	// start a go routine to reload config based on file watcher events
+	if a.baseDeps.RuntimeConfig.AutoReloadConfig {
+		if a.WatchedFiles != nil {
+			w, err := config.NewFileWatcher(a.WatchedFiles, a.baseDeps.Logger)
+			if err != nil {
+				a.baseDeps.Logger.Error("error loading config", "error", err)
+			}
+			a.FileWatcher = w
+			go func() {
```

Comment:
What's going to stop this goroutine?

Reply:
The for loop on line 704 would end if the `EventsCh` is closed, which happen when we call `Stop` on the `FileWatcher`.

That said, I'm still missing some code in line 3724 related to this (stop the watcher and create a new routine).